### PR TITLE
support multi direction

### PR DESCRIPTION
**Pull request by sculove**
## Issue
#462, #466 

## Details
- support multi direction.
- infiniteGrid can use prepend itmes at the first time.
- correct croppedCount in layoutComplete event.

## Preferred reviewers
@netil

### CODE REVIEW

File: src/infiniteGrid.js [hunk 1]
```diff
@@ -262,35 +259,79 @@ eg.module("infiniteGrid", ["jQuery", eg, window, document], function($, ns, glob
...
-				v.isAppend = true;
-				return v;
-			}));
-			this._waitResource(items, isRefresh);
+		layout: function(isRelayout, _addItems, _options) {
+			var options = $.extend({
+				isAppend: true,
+				removedCount: 0
+			}, _options);
+			isRelayout = typeof isRelayout === "undefined" ? true : isRelayout;
```
Comment (netil) [inline src/infiniteGrid.js]: Comparison return boolean, so it doesn't need to explicitly declare boolean.

```js
isRelayout = typeof isRelayout === "undefined" || isRelayout;
```

File: src/infiniteGrid.js [hunk 2]
```diff
@@ -498,20 +518,35 @@ eg.module("infiniteGrid", ["jQuery", eg, window, document], function($, ns, glob
...
+				this._syncCols(false);	// for prepending
 			} else {
-				targets = this.items.splice(idx, this.items.length - idx);
+				targets = idx === this.items.length ? this.items.splice(0) : this.items.splice(idx, this.items.length - idx);
+				this._syncCols(true);	// for appending;
 			}
 
+			// console.log("sync", "<---", this._prependCols);
+			// console.log("sync", "--->", this._appendCols);
+
```
Comment (netil) [inline src/infiniteGrid.js]: remove unnecessary comments

File: src/infiniteGrid.js (hunk 1 above)
Comment (sculove) [inline src/infiniteGrid.js]: Good I will apply it

Comment (sculove) [inline src/infiniteGrid.js]: I apply #468 pr.

File: src/infiniteGrid.js (hunk 2 above)
Comment (sculove) [inline src/infiniteGrid.js]: I apply #468 pr.

**Pull request closed without merging**